internal/controller/pipelinerollout_controller.go

```diff
@@ -646,28 +637,45 @@ func (r *PipelineRolloutReconciler) processPipelineStatus(ctx context.Context, p
 
 	numaLogger.Debugf("pipeline status: %+v", pipelineStatus)
 
+	r.setChildResourcesHealthCondition(pipelineRollout, existingPipelineDef, &pipelineStatus)
+	r.setChildResourcesPauseCondition(pipelineRollout, existingPipelineDef, &pipelineStatus)
+
+	return nil
+}
+
+func (r *PipelineRolloutReconciler) setChildResourcesHealthCondition(pipelineRollout *apiv1.PipelineRollout, pipeline *kubernetes.GenericObject, pipelineStatus *kubernetes.GenericStatus) {
+
 	pipelinePhase := numaflowv1.PipelinePhase(pipelineStatus.Phase)
 	pipelineChildResourceStatus, pipelineChildResourceReason := getPipelineChildResourceHealth(pipelineStatus.Conditions)
 
-	if pipelineChildResourceReason == "Progressing" {
+	if pipelineChildResourceReason == "Progressing" || !pipelineObservedGenerationCurrent(pipeline.Generation, pipelineStatus.ObservedGeneration) {
 		pipelineRollout.Status.MarkChildResourcesUnhealthy("Progressing", "Pipeline Progressing", pipelineRollout.Generation)
-	} else if pipelinePhase == numaflowv1.PipelinePhaseFailed || pipelineChildResourceStatus == "False" {
-		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Failed", pipelineRollout.Generation)
+	} else if pipelinePhase == numaflowv1.PipelinePhaseFailed {
+		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Phase=Failed", pipelineRollout.Generation)
+	} else if pipelineChildResourceStatus == "False" {
+		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Failed, Pipeline Child Resource(s) Unhealthy", pipelineRollout.Generation)
```

Comment:
As far as I can tell this PR didn't introduce any changes for the reason of the condition, right? @dpadhiar can you make sure argocd changes is up to date with latest condition definition? Thanks!

Reply:
This shouldn't make any change for reason of condition as [the lua scripts also check that when the generations are mismatched](https://github.com/argoproj/argo-cd/blob/d2231577c7f667d86bd0aa9505f871ecf1fde2bb/resource_customizations/numaplane.numaproj.io/PipelineRollout/health.lua#L17), we set the Rollout as progressing. This should be good!